lib/Sema/ConstraintSystem.cpp

```diff
@@ -1694,7 +1694,14 @@ void ConstraintSystem::addOverloadSet(ArrayRef<Constraint *> choices,
     return;
   }
 
-  addDisjunctionConstraint(choices, locator, ForgetChoice);
+  auto *disjunction =
+      Constraint::createDisjunction(*this, choices, locator, ForgetChoice);
+  addUnsolvedConstraint(disjunction);
```

Comment:
This seems weird because we add disjunction just to retire it right away, shouldn't it be check and add instead?

Reply:
@xedin Unfortunately `filterDisjunction` currently requires it to be in the inactive list.

Reply:
Oh okay then...